parser/prism/Translator.cc

```diff
@@ -737,9 +741,33 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
                         supportedBlock = true;
                     }
                 } else {
-                    // `PM_BLOCK_ARGUMENT_NODE` is not supported yet.
+                    ENFORCE(PM_NODE_TYPE_P(prismBlock, PM_BLOCK_ARGUMENT_NODE)); // the `&b` in `a.map(&b)`
+
+                    auto *bp = down_cast<pm_block_argument_node>(prismBlock);
+
                     blockPassNode = translate(prismBlock);
-                    supportedBlock = false;
+
+                    if (bp->expression) {
+                        blockPassArgIsSymbol = PM_NODE_TYPE_P(bp->expression, PM_SYMBOL_NODE);
+
+                        if (blockPassArgIsSymbol) {
+                            supportedBlock = false; // TODO: Implement symbol procs
+                        } else {
+                            auto blockPassArgNode = translate(bp->expression);
+
+                            if (supportedCallType && hasExpr(blockPassArgNode)) {
+                                blockPassArg = blockPassArgNode->takeDesugaredExpr();
```

Comment:
It's important that we only do this if `supportedCallType` is true, because this "take" is a destructive operation that ruins the wq tree